src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -892,7 +892,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         }
 
         def cantAdapt =
-          if (context.implicitsEnabled) MissingArgsForMethodTpeError(tree, meth)
+          if (context.implicitsEnabled || context.enrichmentEnabled) MissingArgsForMethodTpeError(tree, meth)
```

Comment:
I never really figured out what's going on here, but my ramblings from https://github.com/scala/scala/pull/6042/files/eec156250d22cdc97dbeb0c66263ff3da74b1268#diff-4eab1aad4533a31c10565971e90f73ea point towards this condition checking that we're not currently type checking a pattern. This change would reverse that. 

In any case (ha!), reporting an error is progress. However, let's provide a better suggestion than eta expansion, which is not allowed in a pattern position.

Reply:
Also, "applicable error message" ha!

Reply:
I tried to look back in history to see when/why the conditional was introduced and failed.

Reply:
OK, I'll read Adriaan's thread and add some LOC. It means I'll lose my scala center certificate for shortest fix of shortest crasher.